src/qinfer/derived_models.py

```diff
@@ -380,6 +381,128 @@ def fisher_information(self, modelparams, expparams):
         )
         return two_outcome_fi * expparams['n_meas']
 
+class GaussianHyperparameterizedModel(DerivedModel):
+    """
+    Model representing a two-outcome model viewed through samples
+    from one of two distinct Gaussian distributions. This model adds four new
+    model parameters to its underlying model, respectively representing the
+    mean outcome conditioned on an underlying 0, the mean outcome conditioned
+    on an underlying 1, and the variance of outcomes conditioned in each case.
+
+    :param qinfer.abstract_model.Model underlying_model: An instance of a two-
+        outcome model to be viewed through Gaussian distributions.
+    """
+
+    def __init__(self, underlying_model):
+        super(GaussianHyperparameterizedModel, self).__init__(underlying_model)
+
+        if not (underlying_model.is_n_outcomes_constant and underlying_model.n_outcomes(None) == 2):
+            raise ValueError("Decorated model must be a two-outcome model.")
+
+    ## PROPERTIES ##
+
+    @property
+    def decorated_model(self):
+        # Provided for backcompat only.
+        return self.underlying_model
+
+    @property
+    def modelparam_names(self):
+        return self.underlying_model.modelparam_names + [
+            r'\mu_0', r'\mu_1',
+            r'\sigma_0^2', r'\sigma_1^2'
+        ]
+    
+    @property
+    def n_modelparams(self):
+        return len(self.modelparam_names)
+    
+    ## METHODS ##
+    
+    def domain(self, expparams):
+        return [RealDomain()] * len(expparams)
```

Comment:
``domain()`` Should support "in the case where ``n_outcomes_constant`` is ``True``, ``None`` should be a valid input".

This requirement is a side-effect of my poor code design IMO, however, we should stick to it until something smoother comes along.

Reply:
Ah, good catch, thank you. I'll go on and fix that, then.